fil_fungible_token/src/token/mod.rs

```diff
@@ -176,28 +197,32 @@ where
     /// Returns the balance associated with a particular address
     ///
     /// Accounts that have never received transfers implicitly have a zero-balance
-    pub fn balance_of(&self, holder: ActorID) -> Result<TokenAmount> {
+    pub fn balance_of(&self, holder: &Address) -> Result<TokenAmount> {
         // Load the HAMT holding balances
+        let holder = self.resolve_to_id_addr(holder)?;
```

Comment:
Use get_id_addr here – we don't want to create the actor. If it doesn't have an ID address it can't have a balance.